### PR TITLE
Display functional context menu when clicking on a contact chip

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/RecipientFieldView.kt [hunk 1]
```diff
@@ -46,14 +51,37 @@ class RecipientFieldView @JvmOverloads constructor(
...
     private val binding by lazy { ViewRecipientFieldBinding.inflate(LayoutInflater.from(context), this, true) }
-
     private var contactAdapter: ContactAdapter? = null
+
+    private var contactMap: Map<String, Map<String, MergedContact>> = emptyMap()
     private val recipients = mutableSetOf<Recipient>()
+    private lateinit var popupRecipient: Recipient
+    private var popupDeletesTheCollapsedChip = false
+
+    private val popupMaxWidth by lazy { 300.toPx() }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/RecipientFieldView.kt:61]: Magic value.

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt [hunk 2]
```diff
@@ -160,6 +158,7 @@ class NewMessageFragment : Fragment() {
             onAutoCompletionToggledCallback = { hasOpened -> toggleAutoCompletion(TO, hasOpened) },
             onContactAddedCallback = { newMessageViewModel.addRecipientToField(it, TO) },
             onContactRemovedCallback = { newMessageViewModel.removeRecipientFromField(it, TO) },
+            onCopyContactAddressCallback = { copyRecipientEmailToClipboard(it) },
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt:161]:
```suggestion
            onCopyContactAddressCallback = ::copyRecipientEmailToClipboard,
```

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt [hunk 3]
```diff
@@ -169,6 +168,7 @@ class NewMessageFragment : Fragment() {
             onAutoCompletionToggledCallback = { hasOpened -> toggleAutoCompletion(CC, hasOpened) },
             onContactAddedCallback = { newMessageViewModel.addRecipientToField(it, CC) },
             onContactRemovedCallback = { newMessageViewModel.removeRecipientFromField(it, CC) },
+            onCopyContactAddressCallback = { copyRecipientEmailToClipboard(it) },
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt:171]:
```suggestion
            onCopyContactAddressCallback = ::copyRecipientEmailToClipboard,
```

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt [hunk 4]
```diff
@@ -177,6 +177,7 @@ class NewMessageFragment : Fragment() {
             onAutoCompletionToggledCallback = { hasOpened -> toggleAutoCompletion(BCC, hasOpened) },
             onContactAddedCallback = { newMessageViewModel.addRecipientToField(it, BCC) },
             onContactRemovedCallback = { newMessageViewModel.removeRecipientFromField(it, BCC) },
+            onCopyContactAddressCallback = { copyRecipientEmailToClipboard(it) },
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt:180]:
```suggestion
            onCopyContactAddressCallback = ::copyRecipientEmailToClipboard,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import com.infomaniak.mail.data.models.correspondent.Correspondent
+import com.infomaniak.mail.data.models.correspondent.MergedContact
+import com.infomaniak.mail.data.models.correspondent.Recipient
+import com.infomaniak.mail.databinding.ViewAvatarNameEmailBinding
+import com.infomaniak.mail.utils.UiUtils.fillInUserNameAndEmail
+
+class AvatarNameEmailView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt:36]:
```suggestion
    defStyleAttr: Int = 0,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by JorisBodin — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**